docs/ROADMAP.md

```diff
@@ -1,3 +1,100 @@
+## Q4 2019 Roadmap
+
+### Overview
+
+In Q4 2019, here are the stories we would like to focus on:
+
+- wrap up on authentication and authorization stories
+- continue with `Inclusion of Related Models` epic
+- add more support for declarative support - From model definition to REST API
+  with no custom repository/controller classes
+- revisit the migration story targeting to have a migration guide
+
+### Scope
+
+#### Authentication
+
+- Token based authentication in API Explorer (Spike done in Q3)
+- [Allow `@authenticate` at class level](https://github.com/strongloop/loopback-next/issues/2460)
+- Refactoring, https://github.com/strongloop/loopback-next/issues/1996 and
+  https://github.com/strongloop/loopback-next/issues/2246
+
+#### Authorization
+
+- Update authorization tutorial
+  https://github.com/strongloop/loopback-next/issues/3694
+- Add authorization to shopping example,
+  https://github.com/strongloop/loopback-next/issues/3695
+
+#### Inclusion of related models
+
+Finish the MVP scope as tracked by Epic https://github.com/strongloop/loopback-next/issues/1352
+    
+- Reject create/update requests when data contains navigational properties https://github.com/strongloop/loopback-next/issues/3439
+- Add inclusion resolvers to lb4 relation CLI https://github.com/strongloop/loopback-next/issues/3451
+- Verify relation type in resolve{Relation}Metadata https://github.com/strongloop/loopback-next/issues/3440
+- Run repository tests for PostgreSQL https://github.com/strongloop/loopback-next/issues/3436
+- Run repository tests for Cloudant https://github.com/strongloop/loopback-next/issues/3437
+- Blog post: announce Inclusion of related models https://github.com/strongloop/loopback-next/issues/3452
+
+See the post-MVP Epic for the list of stories out of scope of the initial release: https://github.com/strongloop/loopback-next/issues/3585
+
+#### From model definition to REST API with no custom repository/controller classes
+
+The Epic: https://github.com/strongloop/loopback-next/issues/2036
+
+- Implement TestSandbox.writeTextFile https://github.com/strongloop/loopback-next/issues/3731
+- Improve `defineCrudRestController` to create a named controller class https://github.com/strongloop/loopback-next/issues/3732
+- Add `defineCrudRepositoryClass` - a helper to create a named repository class https://github.com/strongloop/loopback-next/issues/3733
+- Model API booter & builder https://github.com/strongloop/loopback-next/issues/3736
+- Add CrudRestApiBuilder to `@loopback/rest-crud` https://github.com/strongloop/loopback-next/issues/3737
+- Example app showing CrudRestApiBuilder https://github.com/strongloop/loopback-next/issues/3738
+
+Stretch goals:
+- From relation definition to REST API with auto-generated repository/controller classes https://github.com/strongloop/loopback-next/issues/2483
+- From datasource config to Service REST API with no proxy/controller classes https://github.com/strongloop/loopback-next/issues/3717
```

Comment:
> Stretch goals:
> - From relation definition to REST API with auto-generated repository/controller classes >
> - From datasource config to Service REST API with no proxy/controller classes 

Considering how much work we have already added to the Q4 roadmap, I am proposing to remove these two stretch goals. I feel the other items are more important, we should keep some buffer to add new stories discovering along the way, plus it's probably better to wait for feedback on the initial ModelApiBuilder version before extending it with additional features.

@dhmlau @raymondfeng thoughts?

Reply:
+1
We might want to leave some room for making our migration story better.  
Actually would these 2 be hindering the LB3->LB4 migration?